core/math/math_defs.h

```diff
@@ -31,6 +31,9 @@
 #ifndef MATH_DEFS_H
 #define MATH_DEFS_H
 
+#define CMP_EPSILON_32 0.00001
+#define CMP_EPSILON_64 0.000000001
```

Comment:
The ideal epsilon for 64-bit floats is not the same as the epsilon for 32-bit floats squared. It's about `1e-14` or `0.00000000000001`.

Also, the current epsilon value for 32-bit floats could theoretically have an extra zero added since the ideal is about `1e-6`, but it's a risky change to do if something depends on the extra leniency. I'm not sure that I would bother defining `CMP_EPSILON_32` if it's just the same as `CMP_EPSILON`, but maybe this is a good opportunity to introduce a stricter epsilon value for 32-bit floats.

```suggestion
#define CMP_EPSILON_32 0.000001
#define CMP_EPSILON_64 0.00000000000001
```

Reply:
True, it depends how we want to see `CMP_EPSILON`, because it really isn't THE epsilon from `float` and `double` numbers, it's rather some sort of convenience one that got implicitely used everywhere since every decimal number was a float, but because of that it's getting problematic to decide how to fix problems when we actually have large doubles to deal with.